multihost_test/Sanity/test_pam_pwquality.py

```diff
@@ -0,0 +1,146 @@
+
+import pytest
+import subprocess
+import os
+
+
+def exceute_cmd(multihost, command):
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+@pytest.mark.tier1_2
+class TestPamPwquality(object):
+    def test_pam_pwquality1(self, multihost, create_localusers):
+        """
+        :title: Sanity tests for pam_pwquality.so part1
+        :id: e30c75be-eaf9-11eb-9781-845cf3eff344
+        """
+        exceute_cmd(multihost, 'yum install -y expect')
+        assert 'password' in exceute_cmd(multihost,
+                                         "grep system-auth "
+                                         "/etc/pam.d/passwd").stdout_text
+        exceute_cmd(multihost, "echo pass | passwd --stdin local_anuj")
+        exceute_cmd(multihost, "sed -i 's/.*pam_pwquality.*/password  "
```

Comment:
It would be nice to adapt the `conftest.py` file to handle the pam stack changes, this will be specially helpful for the teardown of the test. Something similar to what's being done for the sssd configuration file.

Moreover, this will help you in the long run because I think that many tests are making changes to the pam stack and having all the functionality embedded in a class will make testing easier.

Reply:
"create_localusers" fixture is doing the job . If any test case fails its will revert back all the changes we have done to beginning stage .   The pam stack changes have to be done in the test case itself as in some of the test cases we are changing the file twice in a frequency, which cant be done with fixture.